python/graphstorm/model/node_gnn.py

```diff
@@ -110,15 +110,15 @@ def forward(self, blocks, node_feats, _, labels, input_nodes=None):
             encode_embs = self.comput_input_embed(input_nodes, node_feats)
         else:
             encode_embs = self.compute_embed_step(blocks, node_feats)
-        # TODO(zhengda) we only support node prediction on one node type now
-        assert len(labels) == 1, "We only support prediction on one node type for now."
-        target_ntype = list(labels.keys())[0]
-        assert target_ntype in encode_embs
-        emb = encode_embs[target_ntype]
-        labels = labels[target_ntype]
-        logits = self.decoder(emb)
-        pred_loss = self.loss_func(logits, labels)
-
+        target_ntypes = list(labels.keys())
+        # compute loss for each node type and aggregate per node type loss
+        pred_loss = 0
+        for target_ntype in target_ntypes:
+            assert target_ntype in encode_embs
+            emb = encode_embs[target_ntype]
+            ntype_labels = labels[target_ntype]
```

Comment:
can you check if target_ntype also exists in labels as well?

Reply:
added to new commit